lib/Upcoming_Ride.dart

```diff
@@ -280,3 +88,153 @@ class UpcomingRide extends StatelessWidget {
     );
   }
 }
+
+class RideInfoWidget extends StatelessWidget {
+  const RideInfoWidget({Key key, this.start, this.end}) : super(key: key);
+
+  final start;
+  final end;
+
+  @override
+  Widget build(BuildContext context) {
+    final fromToStyle = TextStyle(
+      color: Colors.grey[500],
+      fontWeight: FontWeight.w500,
+      fontSize: 12,
+    );
+
+    final infoStyle = TextStyle(
+      color: Colors.black,
+      fontWeight: FontWeight.w400,
+      fontSize: 20,
+    );
+
+    return Container(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Text(
+            'From',
+            style: fromToStyle,
+          ),
+          Text(
+            start,
+            style: infoStyle,
+          ),
+          SizedBox(height: 7),
+          Text(
+            'To',
+            style: fromToStyle,
+          ),
+          Text(
+            end,
+            style: infoStyle,
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class DriverInfoWidget extends StatelessWidget {
+  const DriverInfoWidget({Key key, this.driverName, this.driverStatus})
+      : super(key: key);
+
+  final driverName;
+  final driverStatus;
+
+  @override
+  Widget build(BuildContext context) {
+    final driverToStyle = TextStyle(
+      color: Colors.grey[500],
+      fontWeight: FontWeight.w500,
+      fontSize: 14,
+    );
+
+    final rideStatusStyle = TextStyle(
+      color: Colors.black,
+      fontWeight: FontWeight.w400,
+      fontSize: 20,
+    );
+
+    return Row(
+      children: <Widget>[
+        Icon(Icons.phone),
+        SizedBox(
+          width: 10.0,
+        ),
+        Container(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              Text(
+                driverName,
+                style: driverToStyle,
+              ),
+              Text(
+                driverStatus,
+                style: rideStatusStyle,
+              )
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class UpcomingRideCard extends StatelessWidget {
+  const UpcomingRideCard(
+      {Key key,
+      this.rideStatus,
+      this.timeDateWidget,
+      this.rideInfoWidget,
+      this.driverInfoWidget})
+      : super(key: key);
+
+  final Widget rideStatus;
+  final Widget timeDateWidget;
+  final Widget rideInfoWidget;
+  final Widget driverInfoWidget;
+//  final Widget driverInfoWidget;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
```

Comment:
I think you should wrap a SafeArea around this Container, since right now on my end there is bottom overflow. That way, it allows all devices to prevent this overflow and not just yours.

Reply:
Looks good to me now, Helen's changes fixed this.